### ISSUE TITLE
Rails command doesn't work with version 5.5.1 - Apple M1

### ISSUE THREAD

**Issue by fabiendavy** [puma/puma#2734]
**Describe the bug**
After updating my puma gem from `5.3.1` to `5.5.1` the `rails` command doesn't work anymore

**To Reproduce**
```
bundle
rails c
```

**Error**
```
/Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/activesupport-6.0.4.1/lib/active_support/dependencies.rb:324:in `require': dlopen(/Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/puma-5.5.2/lib/puma/puma_http11.bundle, 9): no suitable image found.  Did find:
	/Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/puma-5.5.2/lib/puma/puma_http11.bundle: mach-o, but wrong architecture
	/Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/puma-5.5.2/lib/puma/puma_http11.bundle: mach-o, but wrong architecture - /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/puma-5.5.2/lib/puma/puma_http11.bundle (LoadError)
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/activesupport-6.0.4.1/lib/active_support/dependencies.rb:324:in `block in require'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/activesupport-6.0.4.1/lib/active_support/dependencies.rb:291:in `load_dependency'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/activesupport-6.0.4.1/lib/active_support/dependencies.rb:324:in `require'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/puma-5.5.2/lib/puma.rb:13:in `<top (required)>'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/bundler-2.2.29/lib/bundler/runtime.rb:60:in `require'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/bundler-2.2.29/lib/bundler/runtime.rb:60:in `block (2 levels) in require'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/bundler-2.2.29/lib/bundler/runtime.rb:55:in `each'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/bundler-2.2.29/lib/bundler/runtime.rb:55:in `block in require'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/bundler-2.2.29/lib/bundler/runtime.rb:44:in `each'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/bundler-2.2.29/lib/bundler/runtime.rb:44:in `require'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/bundler-2.2.29/lib/bundler.rb:174:in `require'
	from /Users/fabien/code/fabiendavy/workelo-app/config/application.rb:21:in `<top (required)>'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/spring-2.1.1/lib/spring/application.rb:92:in `require'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/spring-2.1.1/lib/spring/application.rb:92:in `preload'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/spring-2.1.1/lib/spring/application.rb:157:in `serve'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/spring-2.1.1/lib/spring/application.rb:145:in `block in run'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/spring-2.1.1/lib/spring/application.rb:139:in `loop'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/spring-2.1.1/lib/spring/application.rb:139:in `run'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/gems/2.6.0/gems/spring-2.1.1/lib/spring/application/boot.rb:19:in `<top (required)>'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/fabien/.rbenv/versions/2.6.8/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from -e:1:in `<main>'
```

**Expected behavior**
Open the rails console or server

**Desktop (please complete the following information):**
OS: MacBook Air Apple chips M1

**Comment by joshuapinter** [comment]
Getting the same thing on 5.5.2. We have a M1 Mac in our CI but (some of) our development machines are still on the Intel chipset. Thoughts?

**Comment by nateberkopec** [comment]
Even though this isn't a puma issue, leaving open in case someone figures out the local environment problem here.

**Comment by alexeevit** [comment]
I couldn't reproduce the problem on my laptop with M1 using the same ruby and puma versions.

> I tried to uninstall it then reinstall with arch -x86_64 gem install puma and bundle but I still get the same error

Most likely you use install gem for the wrong architecture. M1 is based on the arm64 but you install for x86_64.

**Comment by fabiendavy** [comment]
I've got some news that solved my problem.
The thing is that my homebrew wasn't well installed. I mean it was install like if my Mac had an Intel Chip.
So I removed my whole homebrew and rbenv to reinstall them with the correct architecture.

I followed all the steps given by Richard (1st answer) of this [question](https://stackoverflow.com/questions/66128681/ruby-build-failed-macos-11-2-using-ruby-build-20210119-mac-big-sur)
